PurchasesCoreSwift/Purchasing/OfferingsFactory.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  OfferingsFactory.swift
+//  PurchasesCoreSwift
+//
+//  Created by César de la Vega on 7/13/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+// TODO (Post-migration): Remove @objc and make it internal again.
+@objc(RCOfferingsFactory) public class OfferingsFactory: NSObject {
+
+    @objc public func createOfferings(withProducts products: [String: SKProduct],
+                                      data: [String: Any]) -> Offerings? {
+        guard let offeringsData = data["offerings"] as? [[String: Any]] else {
+            return nil
+        }
+
+        let offerings = offeringsData.reduce([String: Offering]()) { (dict, offeringData) -> [String: Offering] in
+            var dict = dict
+            if let offering = createOffering(withProducts: products, offeringData: offeringData) {
+                dict[offering.identifier] = offering
+            }
+            return dict
+        }
+        let currentOfferingID = data["current_offering_id"] as? String
+
+        return Offerings(offerings: offerings, currentOfferingID: currentOfferingID)
+    }
+
+    @objc public func createOffering(withProducts products: [String: SKProduct],
+                                     offeringData: [String: Any]) -> Offering? {
+        guard let offeringIdentifier = offeringData["identifier"] as? String,
+              let packagesData = offeringData["packages"] as? [[String: Any]],
+              let serverDescription = offeringData["description"] as? String else {
+            return nil
+        }
+
+        let availablePackages = packagesData.compactMap { packageData -> Package? in
+            createPackage(withData: packageData, products: products, offeringIdentifier: offeringIdentifier)
+        }
+        if availablePackages.count != 0 {
```

Comment:
Slightly more idiomatic to Swift

```swift
guard availablePackages.count > 0 else {
    return nil
}

return Offering(identifier: offeringIdentifier, serverDescription: serverDescription,availablePackages: availablePackages)
```

Reply:
Alternatively

```swift
guard !availablePackages.isEmpty else {
    return nil
}

return Offering(identifier: offeringIdentifier, serverDescription: serverDescription,availablePackages: availablePackages)
```

as `isEmpty` tends to be more performant as well as for improved readability (though I get that some of the readability improvements is negated by the `!` at the beginning).

Reply:
Good point @aamctustwo, I always forget the existence of `.isEmpty`. I do kinda prefer guards to be positive in nature for readability, but I don't have a strong preference here 😄

Reply:
I went with `!` and `isEmpty`. Too bad Swift doesn't have a `isNotEmpty` like Kotlin does